pkg/operator/ceph/disruption/nodedrain/reconcile.go

```diff
@@ -151,16 +164,24 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 		// a new object is going to be created
 		if deploy.ObjectMeta.CreationTimestamp.IsZero() {
 			deploy.Spec.Selector = &metav1.LabelSelector{
-				MatchLabels: deploymentLabels,
+				MatchLabels: selectorLabels,
 			}
 		}
 
-		// update
-		deploy.ObjectMeta.Labels = deploymentLabels
+		// update the deployment labels
+		objectLabels := make(map[string]string, 0)
+		for key, value := range selectorLabels {
+			objectLabels[key] = value
+		}
+		topology, _ := osd.ExtractRookTopologyFromLabels(node.GetLabels())
+		for key, value := range topology {
+			objectLabels[key] = value
```

Comment:
why not add these to the `selectorLabels` map instead of creating a new `objectLabels` map?

Reply:
No important reason. I just had a gut feeling that too many selectors could result in a increase in iterations for the Deployment Controller as it identifies which pods match.

Can change it if necessary.

Reply:
I don't follow. It seems cleaner to just add the topology labels to `selectorLabels` instead of creating a new `objectLabels` map then copying all the labels from `selectorLabels`.

Reply:
@rohantmp Could you update this? Unless I'm missing why it was needed.